### PR TITLE
Use GIR bindings instead of pygtk

### PR DESCRIPTION
**Pull request by raveit65**
origin commit from:
https://github.com/dropbox/nautilus-dropbox/commit/70b9c4481884b26ecd396cf5d5e4522a94e0fd1f

Please test.

### CODE REVIEW

**Comment by monsta** [comment]
The code looks ok, but apparently we need to limit the length of a label.

Before:
dropbox-dialog-before

After:
dropbox-dialog-after

Also the progress bar isn't so large anymore - but I'm not sure if we can change anything, might be GTK+3 thing...

Before:
progress-gtk2

After:
progress-gtk3

**Comment by raveit65** [comment]
I agree with limiting the label width.
With gtk3-inspector
caja-dropbox
But i have no idea to use `gtk_label_set_max_width_chars ()` in python code.
https://developer.gnome.org/gtk3/stable/GtkLabel.html#gtk-label-set-max-width-chars
Here i need some help.

About GtkProgressbar:
This is the normal gtk+-3 progressbar which is controlled by your gtk3 theme.
So it looks like that with pygtk2 something was wrong or they had hard coded the height somewhere.
If i recall correctly there wasn't such a large progressbar in early versions of caja-dropbox.
Maybe this was introduced with porting caja-dropbox to gtk+-3?

**Comment by raveit65** [comment]
Maybe this helps? https://python-gtk-3-tutorial.readthedocs.io/en/latest/label.html

**Comment by raveit65** [comment]
Ok, i will try out https://lazka.github.io/pgi-docs/Gtk-3.0/classes/Label.html#Gtk.Label.set_max_width_chars
`Gtk.Label.set_max_width_chars(n_chars)`

**Comment by raveit65** [comment]
I added `self.label.set_max_width_chars(42)` to line 520, which fixes the large window size.
PR is updated.

File: caja-dropbox.in [hunk 1]
```diff
@@ -323,35 +323,39 @@ class DownloadState(object):
...
+        gi.require_version('Gdk', '3.0')
+        gi.require_version('Gtk', '3.0')
+        from gi.repository import GObject
+        from gi.repository import Gdk
+        from gi.repository import Gtk
+        from gi.repository import Pango
         import webbrowser
 
-        gtk.gdk.threads_init()
+        GObject.threads_init()
```
[the remark this answers is not in the document]
Comment (cwendling) [reply caja-dropbox.in]: The equivalent should be `Gdk.threads_init()`.

[the remark this answers is not in the document]
Comment (raveit65) [reply caja-dropbox.in]: Done, PR is updated.

[the remark this answers is not in the document]
Comment (monsta) [reply caja-dropbox.in]: They say you can drop that line completely - see https://github.com/dropbox/nautilus-dropbox/pull/57#commitcomment-30798147. And PyGObject 3.10.2 is really old :slightly_smiling_face:

**Comment by monsta** [comment]
> If i recall correctly there wasn't such a large progressbar in early versions of caja-dropbox.
Maybe this was introduced with porting caja-dropbox to gtk+-3?

It wasn't ported to GTK+3 until this PR :slightly_smiling_face:

**Comment by raveit65** [comment]
> It wasn't ported to GTK+3 until this PR

Doesn't matter, there wasn't such a huge progressbar with early versions of caja-dropbox.
Some investigations in docs shows me that the text is always outside (next) the progress of the progressbar.
https://developer.gnome.org/gtk3/unstable/GtkProgressBar.html#gtk-progress-bar-set-show-text
And i don't want/know to add a GtkCssProvider in python to get a large progressbar which isn't normal.

File: caja-dropbox.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (raveit65) [reply caja-dropbox.in]: Ok, i will drop it. I guess Martin will never updated ubuntu-14.04 any more :)
https://github.com/dropbox/nautilus-dropbox/pull/57#issuecomment-428318273

[the remark this answers is not in the document]
Comment (raveit65) [reply caja-dropbox.in]: PR is updated again.

**Comment by monsta** [comment]
Yes, looks like the text is always outside the progress bar in GTK+3... we have the same change in our logout/shutdown dialogs.

**Comment by raveit65** [comment]
Thanks for the review.

**Pull request merged**